### ISSUE TITLE
PayPal Express / Pro: Set initial amount to 0 if less than zero

### ISSUE THREAD

**Issue by pippinsplugins** [restrictcontentpro/restrict-content-pro#361]
If the amount sent to PayPal Express / Pro is less than 0 (due to signup fees), an error is thrown at PayPal. We should force it to zero if it's less than zero.

**Comment by tannerm** [comment]
This should be taken care of in #356

**Comment by pippinsplugins** [comment]
:+1:

**Comment by mindctrl** [comment]
When testing #356 as mentioned above, I get this:

![image](https://cloud.githubusercontent.com/assets/1254870/13712244/8b2bffe4-e790-11e5-94b7-d3d502700d6c.png)

When this happened, I was not taken to PayPal at all.

In the backend, there's no $0 payment recorded under Restrict > Payments.

On the member edit screen, the expiration date is correct and the status is correct, but no recurring and no payment.

![image](https://cloud.githubusercontent.com/assets/1254870/13712330/1133c11c-e791-11e5-8bd9-2facfb82731e.png)

**Comment by mindctrl** [comment]
Oh, that was Standard, not sure if that's going to matter or not. Will test Pro/Express.

**Comment by mindctrl** [comment]
Same thing happens with Express. Here's the flow.

If a discount code is used that causes the total to be $0, there is never a subscription created in PayPal. The user stays on the site, and a single-term non-recurring subscription is created. No payment is logged in RCP.

![image](https://cloud.githubusercontent.com/assets/1254870/13713044/ff814d3c-e794-11e5-9bed-19c8fa8b9f19.png)


However, if a discount code is **not** used, and the total is $0, the subscription is created properly and the payment happens.

In summary it looks like the use of a discount code breaks the total recurring / period piece.

**Comment by pippinsplugins** [comment]
Not going to PayPal is actually the intended behavior there, but there are some incorrect things happening:

1. The gateway select field should go away
2. The auto renew option should go away.

@tannerm Could you take a look?

**Comment by tannerm** [comment]
Thanks @pippinsplugins, that's what I was thinking. Yeah, I'll look into those two issues.

**Comment by tannerm** [comment]
@pippinsplugins @mindctrl I just pushed up fixes for the two bugs Pippin outlined.

**Comment by pippinsplugins** [comment]
@mindctrl Could you please test again?

**Comment by pippinsplugins** [comment]
Working for me.
